lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java

```diff
@@ -0,0 +1,182 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.kotlin;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.*;
+
+import com.diffplug.spotless.*;
+
+/** Wraps up [diktat](https://github.com/cqfn/diKTat) as a FormatterStep. */
+public class DiktatStep {
+
+	// prevent direct instantiation
+	private DiktatStep() {}
+
+	private static final String DEFAULT_VERSION = "0.4.0";
+	static final String NAME = "diktat";
+	static final String PACKAGE_DIKTAT = "org.cqfn.diktat";
+	static final String PACKAGE_KTLINT = "com.pinterest.ktlint";
+	static final String MAVEN_COORDINATE = PACKAGE_DIKTAT + ":diktat-rules:";
+
+	public static String defaultVersionDiktat() {
+		return DEFAULT_VERSION;
+	}
+
+	public static FormatterStep create(Provisioner provisioner) {
+		return create(defaultVersionDiktat(), provisioner);
+	}
+
+	public static FormatterStep create(String versionDiktat, Provisioner provisioner) {
+		return create(versionDiktat, provisioner, Collections.emptyMap(), null);
+	}
+
+	public static FormatterStep create(String versionDiktat, Provisioner provisioner, String configPath) {
+		return create(versionDiktat, provisioner, Collections.emptyMap(), configPath);
+	}
+
+	public static FormatterStep create(String versionDiktat, Provisioner provisioner, Map<String, String> userData, String configPath) {
+		return create(versionDiktat, provisioner, false, userData, configPath);
+	}
+
+	public static FormatterStep createForScript(String versionDiktat, Provisioner provisioner, Map<String, String> userData, String configPath) {
+		return create(versionDiktat, provisioner, true, userData, configPath);
+	}
+
+	public static FormatterStep create(String versionDiktat, Provisioner provisioner, boolean isScript, Map<String, String> userData, String configPath) {
+		Objects.requireNonNull(versionDiktat, "versionDiktat");
+		Objects.requireNonNull(provisioner, "provisioner");
+		return FormatterStep.createLazy(NAME,
+				() -> new DiktatStep.State(versionDiktat, provisioner, isScript, userData, configPath),
+				DiktatStep.State::createFormat);
+	}
+
+	static final class State implements Serializable {
+
+		private static final long serialVersionUID = 1L;
+
+		/** Are the files being linted Kotlin script files. */
+		private final boolean isScript;
+		private final String configPath;
```

Comment:
`configPath` should be `FileSignature`, create with `FileSignature.signAsList`, access path with `FileSignature.getOnlyFile()`

The reason is that if this config file changes, the formatting will change too. The `State` class needs to capture every aspect of the formatter or else up-to-date and buildcache behavior will be broken.